### PR TITLE
cpp solution for usaco-623

### PR DESCRIPTION
**Pull request by RedBlueBird**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by RedBlueBird** [comment]
the previous pr closed due to inactivity oops

File: solutions/usaco-623.mdx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+source: USACO Gold 2016 February
+title: Fenced In
+author: Jeffrey Zhang
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_fencedin_gold_feb16.html)
+
+## Solution
+
+The $n$ vertical and $m$ horizontal fences divide the large rectangular pasture into $(n+1)(m+1)$ grids. To help us better comprehend the problem, it is easier for us to visualized the input in terms of graphs: the grids themselves are the nodes, and the fence lengths of each 4 surrounding sides are the edge lengths that can be constructed to their neighboring nodes. For a cow to travel from any node to another, all we need is to construct a tree graph. And to ensure the edges picked are at a minimum, we just need to construct a minimum spanning tree, which can be solved by the [Kruskal's MST algorithm](https://usaco.guide/gold/mst#kruskals).
```
Comment (SansPapyrus683) [inline solutions/usaco-623.mdx:12]:
```suggestion
The $n$ vertical and $m$ horizontal fences divide the large rectangular pasture into $(n+1)(m+1)$ grids. To better understand the problem, it is easier for us to visualize the input in terms of a graph: the grids themselves are the nodes, and the fences of each 4 surrounding sides are the edges that can be constructed. For a cow to travel from any node to another, all we need is to construct a tree. 

To ensure the edges picked are optimal, we construct a minimum spanning tree, which can be solved by [Kruskal's MST algorithm](https://usaco.guide/gold/mst#kruskals).
```

File: solutions/usaco-623.mdx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+typedef pair<int,int> pii;
+#define ff first
+#define ss second
+
+//BeginCodeSnip{Disjoint-Set Union}
+pii find(pii target, vector<vector<pii>>& top){
```
Comment (SansPapyrus683) [inline solutions/usaco-623.mdx:30]: i'd make a class for DSU, up to you though

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
pretty good!

File: solutions/usaco-623.mdx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+author: Jeffrey Zhang
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_fencedin_gold_feb16.html)
+
+## Solution
+
+The $n$ vertical and $m$ horizontal fences divide the large rectangular pasture into $(n+1)(m+1)$ grids. To better understand the problem, it is easier for us to visualize the input in terms of a graph: the grids themselves are the nodes, and the fences of each 4 surrounding sides are the edges that can be constructed. For a cow to travel from any node to another, all we need is to construct a tree. 
+
+To ensure the edges picked are optimal, we construct a minimum spanning tree, which can be solved by [Kruskal's MST algorithm](https://usaco.guide/gold/mst#kruskals).
```
Comment (dongliu0426) [inline solutions/usaco-623.mdx:14]:
```suggestion
To ensure the edges picked are optimal, we construct a minimum spanning tree, which can be solved by [Kruskal's MST algorithm](/gold/mst#kruskals).
```

optional

File: solutions/usaco-623.mdx (hunk 2 above)
Comment (RedBlueBird) [reply solutions/usaco-623.mdx:32]: making an entirely new class might be a bit excessive...

Comment (SansPapyrus683) [reply solutions/usaco-623.mdx:32]: fair enough

**Comment by SansPapyrus683** [comment]
@dongliu0426 you can merge i guess

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**